### PR TITLE
remove unneeded overload for nansum

### PR DESCRIPTION
**Pull request by ngimel**
Per title, fixes a few OpInfo skips

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1486,10 +1486,7 @@
   self: sum_backward(grad, self.sizes(), dim, keepdim)
   result: auto_linear
 
-- name: nansum(Tensor self, *, ScalarType? dtype=None) -> Tensor
-  self: grad.expand(self.sizes()).to(self.scalar_type()) * self.isnan().logical_not()
-
-- name: nansum.dim_IntList(Tensor self, int[1] dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
+- name: nansum.dim_IntList(Tensor self, int[1] dim=[], bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
```
Comment (albanD) [inline tools/autograd/derivatives.yaml:1489]: nit: not really needed to specify the overload.
If not specified, this formula will be used for all overloads.

**Review by albanD — COMMENTED** [review]
So no more full reduction a-la `t.sum()` ?

**Comment by ngimel** [comment]
> So no more full reduction a-la `t.sum()` ?

There is full reduction `x.nansum()`, it works as before, it goes to IntList overload that sets dims to () which is full reduction. 
So, one fewer overload, slightly better functionality (`x.nansum(dim=None)` is now supported whereas previously it wasn't )

**Review by albanD — APPROVED** [review]
Nice cleanup! Thanks!

Not sure what the deal is with the ptl files... Is there some doc about what do do when it happens?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -16203,24 +16203,18 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
-            # FIXME: nansum does not support passing keepdim without passing dim
-            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_default_keepdim'),
             # FIXME: nansum reduces all dimensions when dim=[]
-            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_empty'),
-            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_empty_keepdim'),
-            # FIXME: nansum does not support passing None to dim
-            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_none'),
-            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_none_keepdim'),
-            # FIXME: improve precision
+            DecorateInfo(unittest.expectedFailure, 'TestReductions', 'test_dim_empty'),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:16212]: Nice expectedFailure updates

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -16203,24 +16203,18 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
+            DecorateInfo(unittest.expectedFailure, 'TestReductions', 'test_dim_empty_keepdim'),
+            # FIXME: flaky test so skipped instead of xfailed
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:16214]: Nice comment

**Review by mruberry — APPROVED** [review]
Awesome!

**Comment by ngimel** [comment]
>Not sure what the deal is with the ptl files... Is there some doc about what do do when it happens?
No, and I wasn't able to figure out what to do with IOS, so I fooled it by renaming the dim overload to plain `nansum`,  like it used to be before. 
Mobile fc/bc policy is unclear, so for the most part we are blocked from changing overloads.

**Comment by osalpekar** [comment]
@pytorchbot revert this please

Looks like this breaks functorch (cc @zou3519 @Chillee):
```
Summary: 
pytorch/functorch/functorch/csrc/VmapGeneratedPlumbing.h:10936:46: error: too few arguments to function call, expected 4, have 2
    return at::_ops::nansum::call(self, dtype);
           ~~~~~~~~~~~~~~~~~~~~~~            ^
pytorch/functorch/functorch/csrc/VmapGeneratedPlumbing.h:10951:22: error: no member named 'nansum_dim_IntList' in namespace 'at::_ops'; did you mean 'sum_dim_IntList'?
    return at::_ops::nansum_dim_IntList::call(self, dim, keepdim, dtype);
           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~
pytorch/functorch/functorch/csrc/VmapGeneratedPlumbing.h:10936:46: error: too few arguments to function call, expected 4, have 2
    return at::_ops::nansum::call(self, dtype);
           ~~~~~~~~~~~~~~~~~~~~~~            ^
pytorch/functorch/functorch/csrc/VmapGeneratedPlumbing.h:10951:22: error: no member named 'nansum_dim_IntList' in namespace 'at::_ops'; did you mean 'sum_dim_IntList'?
    return at::_ops::nansum_dim_IntList::call(self, dim, keepdim, dtype);
           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~
```

**Comment by zou3519** [comment]
@ngimel, on the reland, could you please you import this to Phabricator and delete all the nansum related code in functorch? After that gets landed on our side we'll fix up nansum in functorch and then re-sync functorch with fbcode

**Comment by ngimel** [comment]
Revert failed on this one. @zou3519 usually you guys are fixing forward the breakages, so can you please do it?

**Comment by zou3519** [comment]
> Revert failed on this one. @zou3519 usually you guys are fixing forward the breakages, so can you please do it?

Oops, I didn't see that comment. That's a good point, we'll forward fix this

**Comment by zou3519** [comment]
We're forward fixing this now, I have an internal diff ready to go.

For the future, @osalpekar, is there a bug somewhere around pytorchbot and reverts that we should fix?

**Comment by ngimel** [comment]
I don't know why revert bot didn't work, but in general, I don't think that "functorch break" is sufficient reason for revert.

**Comment by osalpekar** [comment]
Revert bot likely couldn't cleanly revert due to subsequently merged PR's touching the same files. Fwd-fix sounds good to me, D35971224 is the diff for for this PR, feel free to stack the fwd-fix atop that diff and land both when ready.

**Comment by mruberry** [comment]
> I don't know why revert bot didn't work, but in general, I don't think that "functorch break" is sufficient reason for revert.

I don't believe we support BC/FC for what functorch is doing in that test, so it should expect to get broken. See https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy. I believe this change would be a "change to a module or operator signatures that doesn't impact users", and so is explicitly exempt from being stable.

**Pull request closed without merging**